src/stdlib_stringlist_type.f90

```diff
@@ -0,0 +1,726 @@
+                                                                insert_before_chararray_int,    &
+                                                                insert_before_stringarray_int
+
+        procedure         :: get_string_idx         => get_string_idx_wrap
+        generic, public   :: get                    => get_string_idx
+
+    end type stringlist_type
+
+    !> Version: experimental
+    !>
+    !> Constructor for stringlist
+    !> Returns an instance of type stringlist_type
+    !> [Specifications](../page/specs/stdlib_stringlist_type.html#stringlist_type)
+    interface stringlist_type
+        module procedure new_stringlist
+        module procedure new_stringlist_carray
+    end interface
+
+    !> Version: experimental
+    !> 
+    !> Concatenates stringlist with the input entity
+    !> Returns a new stringlist
+    !> [Specifications](../page/specs/stdlib_stringlist_type.html#append-operator)
+    interface operator(//)
+        module procedure append_char
+        module procedure append_string
+        module procedure prepend_char
+        module procedure prepend_string
+        module procedure append_stringlist
+        module procedure append_carray
+        module procedure append_sarray
+        module procedure prepend_carray
+        module procedure prepend_sarray
+    end interface
+
+    !> Version: experimental
+    !> 
+    !> Compares stringlist for equality with the input entity
+    !> Returns a logical
+    !> [Specifications](../page/specs/stdlib_stringlist_type.html#equality-operator)
+    interface operator(==)
+        module procedure eq_stringlist
+        module procedure eq_stringlist_carray
+        module procedure eq_stringlist_sarray
+        module procedure eq_carray_stringlist
+        module procedure eq_sarray_stringlist
+    end interface
+
+    !> Version: experimental
+    !> 
+    !> Compares stringlist for inequality with the input entity
+    !> Returns a logical
+    !> [Specifications](../page/specs/stdlib_stringlist_type.html#inequality-operator)
+    interface operator(/=)
+        module procedure ineq_stringlist
+        module procedure ineq_stringlist_carray
+        module procedure ineq_stringlist_sarray
+        module procedure ineq_carray_stringlist
+        module procedure ineq_sarray_stringlist
+    end interface
+
+contains
+
+  ! constructor for stringlist_type:
+
+    !> Constructor with no argument
+    !> Returns a new instance of type stringlist 
+    pure function new_stringlist()
+        type(stringlist_type)                           :: new_stringlist
+        type(string_type), dimension(0)                 :: sarray
+
+        new_stringlist = stringlist_type( sarray )
+
+    end function new_stringlist
+
+    !> Constructor to convert chararray to stringlist
+    !> Returns a new instance of type stringlist
+    pure function new_stringlist_carray( array )
```

Comment:
Why is there not a constructor with array of type `string_type`?

Reply:
After size component was discarded it was causing ambiguous interfaces. Which prompted me to remove it but I have added it again. Please have look at the commit 4b69e3d